README.md

```diff
@@ -1,37 +1,116 @@
-# nlpsandbox-website-synapse
+# NLP Sandbox Synapse Website
 
-Pages of the NLP Sandbox website on Synapse
+Pages of the NLP Sandbox website hosted on Synapse
 
-## Working with website pages
+## Overview
 
-### Fetch pages from Synapse
+This repository includes the pages of the [NLP Sandbox website] hosted on
+Synapse.org. This repository is used by the [NLP Sandbox Team] to track changes
+made to the website pages and continuously deploy (CD) them to Synapse.
 
-The commands below were used to grab the pages of the [NLP Sandbox Staging website]
-in markdown format and save them to the directory [wiki].
+Using this reposiotry instead of Synapse wiki editor offers the following
+advantages.
 
-```
-cd wiki
-challengeutils pull-wiki syn22277124
-```
+- More than one user to edit a page at the same time.
+- The community can propose changes using Pull Requests (PR).
+- Robust review process.
+- The source code of the pages (markdown) is automatically scanned by a linter
+  as part of the CI/CD process to improve code quality.
+- Faster page edition as it becomes possible to edit multiple pages at once
+  (e.g. search and replace).
+- A new website can be created based on this GitHub Template, making it easier
+  to re-use and deploy new websites.
 
-### Push pages to Synapse
+## How to use this repository
 
-The commands below pushed the pages in [wiki] to [NLP Sandbox Staging website].
+This repository relies on two Synapse projects:
 
-```
-cd wiki
-challengeutils push-wiki syn22277124
-```
+- Production site (e.g. [NLP Sandbox website])
+- Staging site
 
-## Mirroring staging site to production
+The default branch of this repository, `staging`, is used to automatically
+deploy changes made to this branch to the staging site on Synapse. We recommend
+configuring repositories created from this template to request that commits
+to the branch `staging` must be submitted using PRs and request one or more
+reviewers to accept the changes before merging them to `staging`, which will
+then trigger the deployment to the staging site.
 
-The command below copies the pages in the staging site to the production site.
+After reviewing the pages on the staging site, a release of this GitHub
+repository can be created to mirror the content of the staging site to the
+production site.
 
-```
-challengeutils mirror-wiki syn22277124 syn22277123
-```
+### Create a new Synapse website
+
+Login to Synapse.org and create two Synapse projects, for example named
+
+- `{Project name}`
+- `{Project name} - staging`
+
+Navigate to the staging site and create the structure of the pages under the
```

Comment:
The problem with creating a repository from this particular template is that people have to edit all the `parentId` to their own wiki pages.  This function isn't robust enough to create the wiki pages by linking the parent locally.  

Maybe the workflow could be:

```
challengeutils create-challenge ...
challengeutils pull-wiki
# make edits
challengeutils push-wiki
```

Reply:
Later in the document I explain that the content of wiki should be emptied and that `pull-wiki` should be run. Also when creating the template repository we talk about, we could initially have the wiki folder empty, until the day where we can use `push-wiki` for the initial population of a wiki.